lxd/network/driver_ovn.go

```diff
@@ -4725,6 +4725,84 @@ func (n *ovn) forwardFlattenVIPs(listenAddress net.IP, defaultTargetAddress net.
 	return vips
 }
 
+// triageExternalAddressUsage performs common steps between creating network forwards and load balancers.
+// This includes validating the provided listen address or auto-allocating one if needed.
+func (n *ovn) triageExternalAddressUsage(listenAddressNet *net.IPNet, proposedAddress string) (*net.IPNet, string, error) {
```

Comment:
can we call this `alocateUplinkAddress`?

Also please can you define in the doc block what the input arguments are, as its not obvious to me. Thanks

Reply:
from what I can see in the usage in the next commit, the `listenAddresNet` and `proposedAddress` arguments are the same value in different formats (the requested listen address or empty/undefined for auto allocation) and the return values are the same value in different formats.

This seems quite confusing to me.

Can we accept just a `listenAddresNet  *net.IPNet`  argument here and return `(net.IP, error)`, and then we can update `forward.ListenAddress` as needed by converting the returned `ip.Net` to a string.

Otherwise we are blurring the lines between allocation & validation logic and formatting logic.